zerver/worker/deferred_work.py

```diff
@@ -126,60 +125,72 @@ def failure_processor(event: dict[str, Any]) -> None:
 
                 retry_event(self.queue_name, event, failure_processor)
         elif event["type"] == "realm_export":
-            realm = Realm.objects.get(id=event["realm_id"])
             output_dir = tempfile.mkdtemp(prefix="zulip-export-")
-            export_event = RealmAuditLog.objects.get(id=event["id"])
             user_profile = get_user_profile_by_id(event["user_profile_id"])
-            extra_data = export_event.extra_data
-            if extra_data.get("started_timestamp") is not None:
+            realm = user_profile.realm
+            export_event = None
+
+            if "realm_export_id" in event:
+                export_row = RealmExport.objects.get(id=event["realm_export_id"])
+            else:
+                # Handle existing events in the queue before we switched to RealmExport model.
+                export_event = RealmAuditLog.objects.get(id=event["id"])
+                extra_data = {}
+                if export_event.extra_data is not None:
+                    extra_data = orjson.loads(export_event.extra_data)
+
+                if extra_data.get("export_row_id") is not None:
+                    export_row = RealmExport.objects.get(id=extra_data["export_row_id"])
+                else:
+                    export_row = RealmExport.objects.create(
+                        realm=realm,
+                        type=RealmExport.EXPORT_PUBLIC,
+                        acting_user=user_profile,
+                        date_requested=event["time"],
+                    )
+                    export_event.extra_data = orjson.dumps(
+                        {"export_row_id": export_row.id}
+                    ).decode()
+                    export_event.save(update_fields=["extra_data"])
+
+            if export_row.date_started is not None:
```

Comment:
I think you want a migration to populate the new table from `RealmAuditLog`; you might still need a bit of code here to look up the object properly, but it'd mean all other code paths can just read from `RealmExport` (like how does one have clean `actions` code listing all the exports if one didn't do a migration?).

Reply:
Yeah, this was still a WIP. A lot of other bugs were still there :)